### PR TITLE
Initial Steamworks integration

### PR DESCRIPTION
**Pull request by Atlante45**
- Steamworks added as an external dependency
- SteamVR can ask Interface to quit
- Added sandbox cmd line opt `--contentPath` to load home content from a local directory.
- Added interface cmd line opt `--suppress-settings-reset` that will prevent the settings reset popup from appearing

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c61b8e2d34932d3a28b158ecfcabafe76067211b available at: https://gist.github.com/e6616a6881a2263ef425a8645a326d9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c61b8e2d34932d3a28b158ecfcabafe76067211b available at: https://gist.github.com/e6616a6881a2263ef425a8645a326d9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1403fad04ea26122cddd07d2a13192160929b43d available at: https://gist.github.com/e6616a6881a2263ef425a8645a326d9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8059.dmg

**Comment by Atlante45** [comment]
Test plan:
1) Launch SteamVR, select Vive display plugin:
    - Quitting SteamVR should close Interface
    - With the Vive/Vive Controllers, open the steam overlay and close Interface from there.

2) Download and unpack `http://cachefly.highfidelity.com/home.tgz`
    - Launch the sandbox via command line: 
        `C:\Program Files\High Fidelity - PR8000\server-console\server-console.exe`
    - Notice the sandbox downloads the home content and then starts
    - Delete the directory `AppData\Roaming\High Fidelity - PR8000`
    -  Launch the sandbox via command line: 
      `C:\Program Files\High Fidelity - PR8000\server-console\server-console.exe -- --contentPatch="C:\Path\To\Content"`
    - Notice this time the home content wasn't downloaded and the sandbox starts right away

3) Launch Interface via command line:
     `C:\Program Files\High Fidelity - PR8000\interface.exe`
    - Force crash it (using Developer menu options or using `CTRL + \`)
    - Launch again via the same command line
    - You should get the reset settings popup, if not enable it `Developer>Ask to reset settings` and retry
    - Force crash interface
    - Launch interface via command line:
     `C:\Program Files\High Fidelity - PR8000\interface.exe --suppress-settings-reset`
    - Notice you don't get the reset settings popup

File: libraries/entities/src/EntityTreeElement.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ void EntityTreeElement::elementEncodeComplete(EncodeBitstreamParams& params) con
             // encoud our parent... this might happen.
             if (extraEncodeData->contains(childElement.get())) {
                 EntityTreeElementExtraEncodeData* childExtraEncodeData
-                    = static_cast<EntityTreeElementExtraEncodeData*>(extraEncodeData->value(childElement.get()));
+                    = static_cast<EntityTreeElementExtraEncodeData*>((*extraEncodeData)[childElement.get()]);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTreeElement.cpp]: wasn't this already included in another PR? This really should be in it's own PR since it's not related at all to steam.

File: plugins/openvr/src/OpenVrDisplayPlugin.cpp [hunk 2]
```diff
@@ -128,10 +128,7 @@ void OpenVrDisplayPlugin::resetSensors() {
 
 bool OpenVrDisplayPlugin::beginFrameRender(uint32_t frameIndex) {
     handleOpenVrEvents();
-    if (openVrQuitRequested()) {
```
Comment (ZappoMan) [inline plugins/openvr/src/OpenVrDisplayPlugin.cpp]: did you really mean to remove this code? This feels like potentially a bad merge.

File: libraries/entities/src/EntityTreeElement.cpp (hunk 1 above)
Comment (Atlante45) [inline libraries/entities/src/EntityTreeElement.cpp]: One of them snuck through.

Comment (ZappoMan) [inline libraries/entities/src/EntityTreeElement.cpp]: ok

**Comment by thoys** [comment]
## QA test report

Tested using HTC Vive

1. I am successfully able to quit interface by closing steamVR when the openVR mode is active. :+1: 
2. I am able to load the content from the path. At first I tried with the `Patch` parameter you mentioned, but realized this was ment to be `Path`  :+1: 
3. I was unable to force crash with Ctrl + \   which is not part of this PR, so I did it manually instead. The  `--suppress-settings-reset` option works as desired!   :+1: 

This QA has a pass :+1:
